pyccel/codegen/wrapper/c_to_python_wrapper.py

```diff
@@ -2352,49 +2426,100 @@ def _extract_HomogeneousTupleType_FunctionDefArgument(self, orig_var, collect_ar
 
         return {'body': body, 'args': arg_vars}
 
-    def connect_pointer_targets(self, orig_var, python_res, funcdef, is_bind_c):
+    def _extract_HomogeneousSetType_FunctionDefArgument(self, orig_var, collect_arg, bound_argument,
+            is_bind_c_argument, *, arg_var = None):
         """
-        Get the code to connect pointers to their targets.
+        Extract the C-compatible homogeneous tuple FunctionDefArgument from the PythonObject.
 
-        Get the code to connect pointers to their targets. The connection is done via reference
-        counting to ensure that the target is not cleaned by the garbage collector before the
-        pointer.
+        Extract the C-compatible homogeneous tuple FunctionDefArgument from the PythonObject.
+        The C-compatible argument is extracted from collect_arg which holds a Python
+        object into arg_var.
+
+        The extraction is done by allocating an array and filling the elements with values
+        extracted from the indexed Python tuple in collect_arg.
 
         Parameters
         ----------
-        orig_var : Variable
-            The result of the function being wrapped.
-        python_res : Variable
-            The Python accessible result of the function being wrapped.
-        funcdef : FunctionDef
-            The function being wrapped.
-        is_bind_c : bool
-            True if the code is translated from a C-compatible language. False if the
-            translated code is in C.
+        orig_var : Variable | IndexedElement
+            An object representing the variable or an element of the variable from the
+            FunctionDefArgument being wrapped.
+
+        collect_arg : Variable
+            A variable with type PythonObject* holding the Python argument from which the
+            C-compatible argument should be collected.
+
+        bound_argument : bool
```

Comment:
What's the point of this argument if it is always false?

Reply:
All the `_extract_X_FunctionDefArgument` functions need to have the same arguments. In `_extract_CustomDataType_FunctionDefArgument` I need to know if the argument is a bound argument or not so this is added as an argument even if it is unhelpful for other functions. This is out of scope for this PR but if you have a suggestion for how to get this information in `_extract_CustomDataType_FunctionDefArgument` without passing it as an argument then we should create an issue and fix this in a dedicated PR.

Reply:
Thank you for the clarification! No, I just didn't get it